src/Dimension.ts

```diff
@@ -25,6 +29,24 @@ export default class Dimension {
     this.name = name
     this.coordNames = coordNames // later, we may make it optional
     this.size = size
+
+    const identityKey = this.identityKey()
+    let identity = identityMap.get(identityKey)
+    if (identity == null) {
+      identity = Symbol(name)
+      identityMap.set(identityKey, identity)
+    }
+    this.identity = identity
+  }
+
+  /**
+   * A string used for dimension identification during comparison.
+   * Mapped into symbol on creation to avoid costly string compares.
+   */
```

Comment:
TSDoc style with return would be helpful.
(Could you write an example of an output?)

Reply:
I'm not sure if it's really relevant. The only meaningful quality of this function output is that it's unique for given dimension "shape", so is a basis for equality. I can make it a bit more clear in the doc. Adding any extra information about it will open the doors for unexpected assumptions.

Reply:
I mean `@returns` field, ideally with an example.